# Profitable App Profiles for the App Store and Google Play Markets

**What is the project about?**

This project is an analitycal investigation of the features that induce  users to download free apps and their engagement with in-app ads.  

**What the goal of the project?**

The goal of this project is to analyze data in order to help our developers understand what types of apps are likely to attract users increasing the incomes from both App Store an Play Store. 

## Opening and Extracting Data ##

The first step is to convert these databases into a list of lists. Thus, I will follow the process that I learned. At the same time, I created a function to print the first few rows and find the number of rows and columns of each data set.

In [1]:
## Open function for Apple ##

opened_file= open("AppleStore.csv")
from csv import reader
read_file = reader(opened_file)
apple_data=list(read_file)
apple_data_header=apple_data[0] ##apple database header##
apple_data_list=apple_data[1:]  ##apple database left##



## Open function for Google ##

opened_file= open("googleplaystore.csv")
from csv import reader
read_file = reader(opened_file)
google_data=list(read_file)
google_data_header=google_data[0]  ##google database header## 
google_data_list=google_data[1:]   ##google database left## 



## Definition of the explore function ##

def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n') # adds a new (empty) line after each row

    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))


##Application of the explore_data function in Apple database##
print(apple_data_header)
print("\n")
explore_data(apple_data_list, 0, 4, True)



['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 7197
Number of columns: 16


In [2]:
##Application of the explore data function in Google database##

print(google_data_header)
print("\n")
explore_data(google_data_list, 0, 4, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 10841
Number of columns: 13


If you want to review the databases the project is built upon, you can use the following links.

Apple dspreadsheet:

1. [link] https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps 

Google spreadsheet:

2. [link] https://www.kaggle.com/lava18/google-play-store-apps

## Deleting Wrong Data ##

In order to get accurate results, the data must be prepared through a process called data cleaning, which allows me to remove inadequate data for my study, for instance, duplicate data.
 

In the discussion regarding Google's database, people detected a certain error on row number 10472, where the category of app called 'Life Made WI-Fi Touchscreen Photo Frame' was missing. To the detect that error, I will create a loop that compares the lenght of each row with the length of the header. Upon its detection, the row is deleted.  


In [3]:
print(len(google_data_list)) ## Google list lenght's before the loop##

for row in google_data_list:
    if len(row) != len(google_data_header):
        print(row)
        print(google_data_list.index(row))
del (google_data_list[10472])

print(len(google_data_list)) ## Google list length's after the loop##


10841
['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']
10472
10840


To check that the wrong row has been deleted (and only this has been removed), I printed the lenght of our google list, and effectively, there is one row less in our database (10841-->10840).

## Removing Duplicate Data

### From Google Database ###

Again, in the discussión, people noticed that there was duplicated data in Google's application list. For instance, if we search for *Instagram*, its displays four entries in our database:

In [4]:
for app in google_data_list:
    name=app[0]
    if name == "Instagram":
        print(app)

['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577446', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66577313', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']
['Instagram', 'SOCIAL', '4.5', '66509917', 'Varies with device', '1,000,000,000+', 'Free', '0', 'Teen', 'Social', 'July 31, 2018', 'Varies with device', 'Varies with device']


Then, to extract these duplicate applications, I created an empty list where we can see and quantify the wrong data to delete it. 

In [5]:
duplicate_apps= []
unique_apps= []

for app in google_data_list:
    name=app[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)
        
print("Number of duplicate apps: ", len(duplicate_apps)) ## Duplicate app length's ##
print("\n")
print("Examples of duplicates apps: ", duplicate_apps[:15])


Number of duplicate apps:  1181


Examples of duplicates apps:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack', 'FreshBooks Classic', 'Insightly CRM', 'QuickBooks Accounting: Invoicing & Expenses', 'HipChat - Chat Built for Teams', 'Xero Accounting Software']


Now, I know that there are 1181 duplicated applications in our Google's database. Some of them are shown above. 
However, I will not remove the duplicated data randomly. I will keep those applications with the highest number of reviews, because it means that they are more updated.  

After removing the duplicates, we should have 9,659 rows like this:

In [6]:
print ("Expected length: ", len(google_data_list)-1181)

Expected length:  9659


First, I will create an empty dictionary called 'reviews_max' where we will save the most updated ones (with the highest number of reviews) and non duplicated applications. Then, using a loop and two conditionals, I will add the applications name in our new dictionary. 

In [7]:
reviews_max= {}

for app in google_data_list:
    name=app[0]
    n_reviews= float(app[3])
    
    if name in reviews_max and reviews_max[name] < n_reviews:
        reviews_max[name] = n_reviews
        
    elif name not in reviews_max:
        reviews_max[name]= n_reviews

print ("Length of the updated dictionary: ",len(reviews_max))
        

Length of the updated dictionary:  9659


As we predicted before, the total number of entries in our dictionary is 9659.

In the next step, I will create two new empty lists to separate and remove the duplicated applications from the original list, which will be called "android_clean" now. 
To get this, we must use a for loop where:
1. We isolate the app name to a variable named "name" and we convert the numbers of reviews to float, and assign it to a variable named "m_reviews".
2. Inside the loop, we will establish a conditional to append the entire row to the "android_clean" list and add the name of the app "name" to the "already_added" list. 
    
Finally, to check the result, I will print the actual length of the list "android_clean".

In [8]:
android_clean= [] ## we will store our data cleaned data set ##
already_added=[]  ## we will just store app names ##

for app in google_data_list:
    name=app[0]
    n_reviews=float(app[3])
    
    if (reviews_max[name]== n_reviews) and (name not in already_added):
        android_clean.append(app)
        already_added.append(name)

print("Actual length: ", len(android_clean))
        
    

Actual length:  9659


## Removing non English Application ##

Considering that one of our goals is to analyze data from the apps created for English speakers, we would need to filter the non-English ones and remove them . 


According to the ASCII (American Standard Code for Information Interchange) system, each character has a number associated. Those used in the English alphabet ranges from 0 to 127. 

With this information, I can create a function that detects whether a character belongs to the set of common English characters called "english_character". It contains a for loop with a conditional that examine each app of our database.

In [9]:
def english_character(string):
    for character in string:
        if ord (character) > 127:
            return False
    return True
        

Then, I will use this function over some apps to make sure that it detects if they are English or not: 

In [10]:
print(english_character('Instagram'))
print(english_character('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(english_character('Docs To Go™ Free Office Suite'))
print(english_character('Instachat 😜'))

True
False
False
False


Based on the result, *Instagram* would be the only English app, but that is wrong. The third and the fourth example above are English-apps as well, but my function did not reccognize them because they use a super index and a emoji, respectively. 

If we are going to use the function we have created, we will lose useful data since many English apps will be incorrectly labeled as non-English. To minimize the impact of data loss, I will only remove an app if its name has more than three characters with corresponding numbers falling outside the ASCII range.

Thus, I will change the "english_character" function:

In [11]:
def english_character(string):
    non_ascii = 0
    
    for character in string:
        if ord(character)> 127: 
            non_ascii+=1
    if non_ascii >3:
        return False
    else:
        return True

In [12]:
print(english_character("Docs To Go™ Free Office Suite"))
print(english_character("Instachat 😜"))
print(english_character("爱奇艺PPS -《欢乐颂2》电视剧热播"))

True
True
False


Now that the results are more accurate, I will apply our new function over the Google and the Apple database: 

In [13]:
apple_english=[]
android_english=[]

for app in apple_data_list:
    name=app[1]
    if english_character(name):
        apple_english.append(app)
    
for app in android_clean:
    name=app[0]
    if english_character(name):
        android_english.append(app)
    
print("The actual length of Apple Database is:", len(apple_english))
print("The actual length of Google Database is:", len(android_english))

The actual length of Apple Database is: 6183
The actual length of Google Database is: 9614


With all data sets clean, I will explore both to see how many rows we have remaining for each data set. 

In [14]:
explore_data(apple_english, 0, 4, True)
print("\n")
explore_data(android_english, 0, 4, True)

['284882215', 'Facebook', '389879808', 'USD', '0.0', '2974676', '212', '3.5', '3.5', '95.0', '4+', 'Social Networking', '37', '1', '29', '1']


['389801252', 'Instagram', '113954816', 'USD', '0.0', '2161558', '1289', '4.5', '4.0', '10.23', '12+', 'Photo & Video', '37', '0', '29', '1']


['529479190', 'Clash of Clans', '116476928', 'USD', '0.0', '2130805', '579', '4.5', '4.5', '9.24.12', '9+', 'Games', '38', '5', '18', '1']


['420009108', 'Temple Run', '65921024', 'USD', '0.0', '1724546', '3842', '4.5', '4.0', '1.6.2', '9+', 'Games', '40', '5', '1', '1']


Number of rows: 6183
Number of columns: 16


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketc

So far in the data cleaning process, we have:

1. Removed inaccurate data
2. Removed duplicate app entries
3. Removed non-English apps

## Isolating Free Applications ##

In the next step, I have to isolate only free apps from both database and add them in a two new empty lists, because I only work with English and free applications in my research.  

In [15]:
english_free_google= []
english_free_apple= []

for app in android_english:
    price = app[7]
    if price == "0":
        english_free_google.append(app)
        

for app in apple_english:
    price = app[4]
    if price == "0.0":
        english_free_apple.append(app)
        
print("The actual length od our Google database is:",len(english_free_google))
print("The actual length od our Apple database is:",len(english_free_apple))

The actual length od our Google database is: 8864
The actual length od our Apple database is: 3222


Up to this point, inaccurate data, duplicated app entries, and non_English apps have been removed; and at the same time, free apps have been isolated.

## Validation Strategy ##


To minimize risks and overhead, my validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, we develop it further.
3. If the app is profitable after six months, we build an iOS version of the app and add it to the App Store

Because our end goal is to add the app on both Google Play and the App Store, I need to find app profiles that are successful on both markets.

Now, I am going to identify what columns are necessary to do frecuency tables. I will print the header of both databases and choose the propiated column. 

In [16]:
print(apple_data_header)
print("\n")
print(google_data_header)

['id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


We can see that the main columns (related to genre) for this part of the project are:
- For Apple database: column 12
- For Google database: column 10

# Frecuency Tables #

I will build two functions we can use to analyze the frequency tables:

- One function to generate frequency tables that show percentages.
- Another function we can use to display the percentages in a descending order.

To do that, I will need to make use of the built-in sorted() function. This function takes in an iterable data type (like a list, dictionary, tuple, etc.), and returns a list of the elements of that iterable sorted in ascending or descending order (the reverse parameter controls whether the order is ascending or descending).

But before, I have to transform the dictionary into a list of tuples because "sorted()" function does not work with too well in dictionaries. 

Now, I will create a function for generating frequency tables, and use it in combination with the display_table() function.

In [17]:
def freq_table(dataset, index):
    table={}
    total=0 
    
    for row in dataset:

        total+=1
        value=row[index]
       

        if value in table:
            table[value]+=1
        
        else:
            table[value]= 1
    
    table_percentages={}
    for key in table:
        percentage= (table[key]/total)*100
        table_percentages[key]=percentage 
        
    return table_percentages
    

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])
    
   
        

Once function are defined, I will apply both of them over our data sets and will display the frequency table of the columns "prime_genre", "Genres", and "Category" :


## For Apple Data Set ##


Prime_genre:

In [18]:
display_table(english_free_apple, -5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


As we can see in the prime genre list of Apple dataset, games are the main English free genre with 58% of total apps. Also, entertainment and photo and video represent between 8% and 5% of total genre apps, respectively. Also, with a lower ratio, we can find genres like education, social networking, utilities and sports. 

With these percentages, most of the apps are designed for fun like games, while the practical purposes are more rare. 

However, the fact that fun apps are the most numerous does not also imply that they also have the greatest number of users — the demand might not be the same as the offer.

## For Google Data Set ##

Category:

In [19]:
display_table(english_free_google, 1 )

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

Unlike of Apple Store users, the percentage of category app in Play Store are more equitative.It seems that they more related with practical purpose than fun. However, although family category keeps the first position in this list, many of their apps correspond to games for kids.  

Genres:

In [20]:
display_table(english_free_google, -4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

The most common genres in Play Store are tool with around 9% of all apps, followed by entertainment, education and business with 6%, 5% and 4%, respectively.

For the time being, we will focus on Play Store genres more than categories, because we will provide us more information and can contain some categories. 

# More Popular Apps by Genre between Apple Users #

One way to find out what genres are the most popular (have the most users) is to calculate the average number of installs for each app genre.


The frequency tables we analyzed on the previous screen showed us that the App Store is dominated by apps designed for fun, while Google Play shows a more balanced landscape of both practical and fun apps. Now, we would like to get an idea about the kind of apps with the most users.

Thus, I will need to calculate the average number of user ratings per app genre:

1. Isolate the apps of each genre.
2. Sum up the user ratings for the apps of that genre.
3. Divide the sum by the number of apps belonging to that genre (not by the total number of apps).

In [21]:
genre_apple=freq_table(english_free_apple, -5)

for genre in genre_apple:
    total=0
    len_genre=0
    
    for app in english_free_apple:
        genre_app=app[-5]
        
        if genre_app == genre:
            number_user_rating = float (app[5])
            total+= number_user_rating
            len_genre+=1

    average_number= total/len_genre
    print(genre, ":", average_number)


Social Networking : 71548.34905660378
Photo & Video : 28441.54375
Games : 22788.6696905016
Music : 57326.530303030304
Reference : 74942.11111111111
Health & Fitness : 23298.015384615384
Weather : 52279.892857142855
Utilities : 18684.456790123455
Travel : 28243.8
Shopping : 26919.690476190477
News : 21248.023255813954
Navigation : 86090.33333333333
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Food & Drink : 33333.92307692308
Sports : 23008.898550724636
Book : 39758.5
Finance : 31467.944444444445
Education : 7003.983050847458
Productivity : 21028.410714285714
Business : 7491.117647058823
Catalogs : 4004.0
Medical : 612.0


If we focus on the table above, we see that navigation genre app is the most common between users, because most of people need them in their daily routine to seek information or whathever place that they want to go. However, it is not the first genre when we have analyzed what genre is the 
most downloaded. This is because we only need to download them once, while new games come out every day. 


In the next step, I am going to show what kind of navigation apps are more popular between users. 

In [22]:
for app in english_free_apple:
    if app[-5] == 'Navigation':
        print(app[1],':', app[5])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Google Maps - Navigation & Transit : 154911
Geocaching® : 12811
CoPilot GPS – Car Navigation & Offline Maps : 3582
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5


Waze and Google Maps occupy the first positions with half a million of downloads.

Then, I am going to check what sport apps are more popular between users, because it has one of the highest number of installs. It could be a good niche to invest in a near future.  

In [23]:
for app in english_free_apple:
    if app[-5] == 'Sports':
        print(app[1], ':', app[5])

ESPN: Get scores, news, alerts & watch live sports : 290996
Yahoo Fantasy Sports : 190670
WatchESPN : 159735
The Masters Tournament : 148160
Yahoo Sports - Teams, Scores, News & Highlights : 137951
ESPN Fantasy Football Baseball Basketball Hockey : 64925
CBS Sports App - Sports Scores, News, Stats, Watch : 59639
FOX Sports Mobile : 57500
2016 U.S. Open Golf Championship : 54192
NBC Sports : 47172
NBA : 43682
ESPN Tournament Challenge : 39642
2016 US Open Tennis Championships : 37522
NFL : 27317
MLB.com At Bat : 21830
The Championships, Wimbledon 2016 - Tennis Grand Slam : 20953
DraftKings - Daily Fantasy Golf, Baseball, & More : 20251
Bleacher Report: Sports news, scores, & highlights : 16979
Univision Deportes: Liga MX, MLS, Fútbol En Vivo : 16683
NASCAR MOBILE : 16385
NHL : 15554
FOX Sports GO : 13898
NCAA March Madness Live - Men's College Basketball : 13572
PGA Championship 2016 – Baltusrol Golf Club : 10472
365Scores : 9879
NFL Fantasy Football - Official NFL Fantasy App : 8234
Fa

On this branch, we can see that it exists a great variety because of the big number of different sports that people follow. It can be easy to create an app with ads and high number of visualizations. 

# More Popular Apps by Genre between Android Users #

Previously, we came up with an app profile recommendation for the App Store based on the number of user ratings. We have data about the number of installs for the Google Play market, so I should be able to get a clearer picture about genre popularity. However, the install numbers do not seem precise enough — we can see that most values are open-ended (100+, 1,000+, 5,000+, etc.).

In [24]:
display_table(english_free_google, 5)

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


For instance, we do not know whether an app with 100,000+ installs has 100,000 installs, 200,000, or 350,000. However, I do not need very precise data for our purposes — we only want to find out which app genres attract the most users, and we do not need perfect precision with respect to the number of users.

I am going to leave the numbers as they are, which means that I will consider that an app with 100,000+ installs has 100,000 installs, and an app with 1,000,000+ installs has 1,000,000 installs, and so on. To perform computations, however, I will need to convert each install number from string to float. This means I need to remove the commas and the plus characters, otherwise the conversion will fail and raise an error.

In [25]:
category_table= freq_table(english_free_google, 1)

for category in category_table:
    total= 0
    len_category= 0
    
    for app in english_free_google:
        category_app= app[1]
        
        if category_app== category:
            n_installs= app[5]
            n_installs= n_installs.replace (',','')
            n_installs= n_installs.replace ('+','')
            total+= float(n_installs)
            len_category+= 1
            
    av_number= total / len_category
    print(category, ":", av_number)
        
        


ART_AND_DESIGN : 1986335.0877192982
AUTO_AND_VEHICLES : 647317.8170731707
BEAUTY : 513151.88679245283
BOOKS_AND_REFERENCE : 8767811.894736841
BUSINESS : 1712290.1474201474
COMICS : 817657.2727272727
COMMUNICATION : 38456119.167247385
DATING : 854028.8303030303
EDUCATION : 1833495.145631068
ENTERTAINMENT : 11640705.88235294
EVENTS : 253542.22222222222
FINANCE : 1387692.475609756
FOOD_AND_DRINK : 1924897.7363636363
HEALTH_AND_FITNESS : 4188821.9853479853
HOUSE_AND_HOME : 1331540.5616438356
LIBRARIES_AND_DEMO : 638503.734939759
LIFESTYLE : 1437816.2687861272
GAME : 15588015.603248259
FAMILY : 3695641.8198090694
MEDICAL : 120550.61980830671
SOCIAL : 23253652.127118643
SHOPPING : 7036877.311557789
PHOTOGRAPHY : 17840110.40229885
SPORTS : 3638640.1428571427
TRAVEL_AND_LOCAL : 13984077.710144928
TOOLS : 10801391.298666667
PERSONALIZATION : 5201482.6122448975
PRODUCTIVITY : 16787331.344927534
PARENTING : 542603.6206896552
WEATHER : 5074486.197183099
VIDEO_PLAYERS : 24727872.452830188
NEWS_AND_

As we can see, the main genre for Google users are related with communication. It is a consistent data because of the new technological era that we life right now, where all we need to be connected.  On average, communication apps have the most installs: 38,456,119. 

In [26]:
for app in english_free_google:
    if app[1] == 'COMMUNICATION' and (app[5]=='1,000,000,000+' 
                                      or app[5]=='500,000,000+' 
                                      or app[5]=='100,000,000+'):
        print(app[0],':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

In this table, we can check that WhatsApp, Skype and Hangout apps 
are the most outstanding among all of them. I think that communication is a genre too harness and the competition is too great on this kind of market. It could be very difficult to stand out and get incomes with engaged ads.

However, like in Apple Store, sport genre are popular too between Android users. Below, I am going to extract the name of more popular apps:

In [27]:
for app in english_free_google:
    if app[1] == 'SPORTS' and (app[5]=='5,000,000+' 
                                      or app[5]=='1,000,000+' 
                                      or app[5]=='10,000,000+'):
        print(app[0],':', app[5])

LiveScore: Live Sport Updates : 10,000,000+
Onefootball - Soccer Scores : 10,000,000+
Futbol24 : 1,000,000+
kicker football news : 5,000,000+
Football Live Scores : 5,000,000+
Pro 2018 - Series A and B : 5,000,000+
BeSoccer - Soccer Live Score : 10,000,000+
Sport.pl LIVE : 1,000,000+
The Team - Live Sport: football, tennis, rugby .. : 5,000,000+
MARCA - Sports Leader Diary : 5,000,000+
Sahadan Live Scores : 5,000,000+
Foot Mercato: transfers, results, news, live : 1,000,000+
All Football - Latest News & Videos : 10,000,000+
Eurosport : 10,000,000+
beIN SPORTS TR : 10,000,000+
Top Mercato: football news : 1,000,000+
GollerCepte Live Score : 1,000,000+
Championat : 1,000,000+
Speed Boat Racing : 1,000,000+
La Liga - Spanish Soccer League Official : 10,000,000+
Goal Live Scores : 5,000,000+
Mackolik Live Results : 10,000,000+
WatchESPN : 10,000,000+
CBS Sports Fantasy : 1,000,000+
NBC Sports : 5,000,000+
UFC : 1,000,000+
Telemundo Deportes - Live : 1,000,000+
Golf Channel : 1,000,000+
Gol

Perhaps, sport apps are not as relevant as communication apps but it is kind of market with high demand. It is true that exists a great variety of apps of this genre, but only few of them have news and videos in their content.  

# Conclusion #

In this project, I analyzed data about the App Store and Google Play mobile apps with the goal of recommending an app profile that can be profitable for both markets.

I found that both Android and Apple users use many apps related to news or watching live videos. Many of these app are free to download and have engaged ads with them. Also, it is a kind of market with high demand. 

Perhaps, creating an app that include news and videos from different sports can be a good option and report us great incomes in the near future. 